drivers/flash/flash_npcx_fiu_nor.c

```diff
@@ -278,27 +284,40 @@ static int flash_npcx_nor_erase(const struct device *dev, off_t addr,
 		return -EINVAL;
 	}
 
+	if ((config->min_erase_size != BLOCK_4K_SIZE && config->min_erase_size != BLOCK_64K_SIZE) ||
+	    (config->max_erase_size != BLOCK_4K_SIZE && config->max_erase_size != BLOCK_64K_SIZE)) {
+		return -EINVAL;
+	}
+
 	/* Select erase opcode by size */
 	if (size == config->flash_size) {
 		flash_npcx_uma_cmd_only(dev, SPI_NOR_CMD_WREN);
 		/* Send chip erase command */
 		flash_npcx_uma_cmd_only(dev, SPI_NOR_CMD_CE);
 		return flash_npcx_nor_wait_until_ready(dev);
-	} else if (config->min_erase_size == KB(4)) {
-		opcode = SPI_NOR_CMD_SE;
-	} else if (config->min_erase_size == KB(64)) {
-		opcode = SPI_NOR_CMD_BE;
-	} else {
-		return -EINVAL;
+	}
+
+	if (config->min_erase_size <= BLOCK_64K_SIZE && BLOCK_64K_SIZE <= config->max_erase_size) {
+		erase_block_64k = true;
+	}
+	if (config->min_erase_size <= BLOCK_4K_SIZE && BLOCK_4K_SIZE <= config->max_erase_size) {
+		erase_block_4k = true;
 	}
 
 	while (size > 0) {
 		flash_npcx_uma_cmd_only(dev, SPI_NOR_CMD_WREN);
 		/* Send page/block erase command with addr */
-		flash_npcx_uma_cmd_by_addr(dev, opcode, addr);
-		addr += config->min_erase_size;
-		size -= config->min_erase_size;
-		ret = flash_npcx_nor_wait_until_ready(dev);
+		if (erase_block_64k && (size >= BLOCK_64K_SIZE)) {
+			flash_npcx_uma_cmd_by_addr(dev, SPI_NOR_CMD_BE, addr);
+			addr += BLOCK_64K_SIZE;
+			size -= BLOCK_64K_SIZE;
+			ret = flash_npcx_nor_wait_until_ready(dev);
```

Comment:
Move duplicated `ret = flash_npcx_nor_wait_until_ready(dev);` from line 314 and 319 to line 321?

Reply:
I moved the suggested line out of if else in latest revision, thank you